libraries/rush-lib/src/cli/RushXCommandLine.ts

```
@@ -45,129 +47,139 @@ interface IRushXCommandLineArguments {
-        commandWithArgsForDisplay += ' ' + args.commandArgs.join(' ');
-      }
+      commandWithArgs += ' ' + escapedRemainingArgs.join(' ');
 
-      if (!args.quiet) {
-        console.log(`> ${JSON.stringify(commandWithArgsForDisplay)}\n`);
-      }
+      // Display it nicely without the extra quotes
+      commandWithArgsForDisplay += ' ' + args.commandArgs.join(' ');
+    }
 
-      const packageFolder: string = path.dirname(packageJsonFilePath);
-
-      const exitCode: number = Utilities.executeLifecycleCommand(commandWithArgs, {
-        rushConfiguration,
-        workingDirectory: packageFolder,
-        // If there is a rush.json then use its .npmrc from the temp folder.
-        // Otherwise look for npmrc in the project folder.
-        initCwd: rushConfiguration ? rushConfiguration.commonTempFolder : packageFolder,
-        handleOutput: false,
-        environmentPathOptions: {
-          includeProjectBin: true
-        }
-      });
+    if (!args.quiet) {
+      console.log(`> ${JSON.stringify(commandWithArgsForDisplay)}\n`);
+    }
 
-      if (exitCode > 0) {
-        console.log(colors.red(`The script failed with exit code ${exitCode}`));
+    const packageFolder: string = path.dirname(packageJsonFilePath);
+
+    const exitCode: number = Utilities.executeLifecycleCommand(commandWithArgs, {
+      rushConfiguration,
+      workingDirectory: packageFolder,
+      // If there is a rush.json then use its .npmrc from the temp folder.
+      // Otherwise look for npmrc in the project folder.
+      initCwd: rushConfiguration ? rushConfiguration.commonTempFolder : packageFolder,
+      handleOutput: false,
+      environmentPathOptions: {
+        includeProjectBin: true
       }
+    });
 
-      process.exitCode = exitCode;
-    } catch (error) {
-      console.log(colors.red('Error: ' + (error as Error).message));
+    if (exitCode > 0) {
+      console.log(colors.red(`The script failed with exit code ${exitCode}`));
     }
```

Nearly the entire contents of this internal method were wrapped in the try/catch, so I moved that up to the caller.  Again, this makes the internal version easier to test.   

I think the process.exitCode should probably move up to the caller as well.

---

If so, have `exitCode` be the return value.

---

OK, what I did instead is replace the returns with throws (where there is an error).

---

@octogonz , @dmichon-msft Please mark this thread resolved if you are happy with the decision to throw instead of returning exit codes.

---

As long as you tested it and it works, I'm fine with it.